### ISSUE TITLE
Refactor `UiaTextRange`

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/terminal#3895]
There's a lot of duplicate code in `UiaTextRange`. This should be refactored in such a way to...
- reduce duplicate code
- separate navigation type by code files (i.e.: cell nav, word nav, line nav, etc...)

#1993 could be a part of this.
